source/Calamari/Kubernetes/KubernetesManifestNamespaceResolver.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using YamlDotNet.RepresentationModel;
+
+namespace Calamari.Kubernetes
+{
+    public interface IKubernetesManifestNamespaceResolver
+    {
+        string ResolveNamespace(YamlMappingNode rootManifestNode, IVariables variables);
+    }
+
+    public class KubernetesManifestNamespaceResolver : IKubernetesManifestNamespaceResolver
```

Comment:
This is the new class that contains all the logic for resolving a namespace for a given manifest root node.